event_model/schemas/stream_datum.json

```diff
@@ -0,0 +1,40 @@
+{
+    "properties": {
+        "datum_kwargs": {
+            "type": "object",
+            "description": "Arguments to pass to the Handler to retrieve one quanta of data"
+        },
+        "stream_resource": {
+            "type": "string",
+            "description": "The UID of the Stream Resource to which this Datum belongs"
+        },
+        "uid": {
+            "type": "string",
+            "description": "Globally unique identifier for this Datum. A suggested formatting being '<stream_resource>/<stream_name>/<block_id>"
+        },
+        "stream_name": {
+            "type": "string",
+            "description": "The name of the stream that this Datum is providing a block of."
+        },
+        "block_id": {
+            "type": "integer",
+            "description": "The order in the stream of this block of data. This must be contiguous for a given stream."
+        },
+        "event_offset": {
+            "type": "integer",
+            "description": "The sequence number of the first event in this block. This increasing value allows the presence of gaps."
+        }
```

Comment:
We need some concept of `event_count` to tell us how many Events' worth of data this is going to give us.

Reply:
Addressed initially in https://github.com/bluesky/event-model/pull/222#:~:text=count%20to%20stream-,9c6b8da,-Merge%20state

Reply:
If we allow each datum to have a variable number of events in it (which is something we need for many of the flying use cases, it is much simpler to bundle and emit "what ever I just got in the last second" than "bundle and emit when I have _exactly_ 50 events" and we will in general have less events in the final datum), than if the user says "I want events `[10_000:15_000]`" then if all we have is block index then we have to iterative count from the beginning.  On the other hand if we have the offsets in each document we can say "dear database, please give me all datum with offsets in this range and the one with the highest block ID that is less that start" which should be much more efficient.

We ales need this when you are working in a context where you need to pre-allocate space for the data before you load it.